.eslintrc.js

```diff
@@ -51,6 +51,12 @@ module.exports = {
     'react/display-name': 'off',
     'import/no-named-as-default': 'off',
     'import/no-cycle': ['error', { maxDepth: '∞' }],
+    'prettier/prettier': [
+      'error',
+      {
+        endOfLine: 'auto',
+      },
+    ],
```

Comment:
is this expected?

Reply:
- Prettier and Linting were failing on the Windows Environment with an error saying to delete CR rule from the prettier config. I could not find any config related to this. So, upon googling what the issue is and the preferred solution, this was what was recommended.

- https://prettier.io/docs/en/options.html#end-of-line

- As per docs, the default rule is 'LF' which is widely accepted across all platforms and prevents any line ending inconsistencies, but it did not resolve the issue on my end. So, I've set the rule to 'auto' which was previously widely accepted.

------------------------------------------------------------------------
I did find another solution, after diving deep into this. But I'm unsure about its implementation. Sharing the solution below for your reference:

> The change from LF (Line Feed) to CRLF (Carriage Return + Line Feed) line endings when you open a .js file in Visual Studio Code on Windows might be due to the default line ending settings in your environment. This can potentially cause issues, especially if you collaborate with others or work on projects with a specific line ending requirement.

> The issue arises because Windows traditionally uses CRLF line endings, while many other platforms (like Linux and macOS) use LF. When you work on a project with mixed line endings, it can lead to inconsistencies and potential version control problems.

> Configuring line endings in your project's `.gitattributes` file is essential for ensuring consistency when using Git for version control. Here's how to do it:

1. Open your project's `.gitattributes` file if it already exists. If it doesn't exist, you can create it in the root of your Git repository.

2. Inside the `.gitattributes` file, you can specify the line endings for specific file types or directories. Here's the syntax:

   ```
   pattern text=lf
   ```

   - `pattern` is a file pattern or directory that you want to apply the line-ending settings to.
   - `text=lf` specifies that the line endings should be converted to LF (Unix-style).

   For example, to set LF line endings for all `.js` files, you can add the following line:

   ```
   *.js text=lf
   ```

3. Save and commit your `.gitattributes` file to your Git repository.

Once you've configured the `.gitattributes` file, Git will use the specified line endings when checking out and committing files, ensuring consistent line endings for everyone working on the project. This helps prevent line ending issues when collaborating on cross-platform projects.

Reply:
@anuraghazra I just came across the `.editorconfig` file that you've created for resolving this issue, but it seems to be not working as intended on Windows Environment. Any suggestions on how this can be resolved ?

Reply:
@TejasShekar can you try this https://stackoverflow.com/a/48523398?

install it globally with `yarn add --global editorconfig`

Reply:
Setting it to `auto` might let OS decide type of line ends so it will end up creating different line ends on different operating systems ;__;

Reply:
Ahh, right. I was going through the editorconfig docs in the morning and its mentioned that VSCode does not natively support this file and needs a plugin. Will install that and remove this prettier rule and update PR.